tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp

```diff
@@ -75,77 +129,121 @@ operation::ProgramWithCallbacks MorehAdam::create_program(
     const std::vector<Tensor>& input_tensors,
     const std::vector<std::optional<const Tensor>>& optional_input_tensors,
     std::vector<Tensor>& output_tensors) const {
-    const auto& param = input_tensors.at(0);
+    const auto& param_in = input_tensors.at(0);
     const auto& grad = input_tensors.at(1);
-    const auto& exp_avg = input_tensors.at(2);
-    const auto& exp_avg_sq = input_tensors.at(3);
-    const auto& max_exp_avg_sq = optional_input_tensors.at(0);
+    const auto& exp_avg_in = input_tensors.at(2);
+    const auto& exp_avg_sq_in = input_tensors.at(3);
+    const auto& max_exp_avg_sq_in = optional_input_tensors.at(0);
+
+    const auto& param_out = output_tensors.at(0);
+    const auto& exp_avg_out = output_tensors.at(1);
+    const auto& exp_avg_sq_out = output_tensors.at(2);
+    const auto& max_exp_avg_sq_out =
+        given_max_exp_avg_sq_in ? std::make_optional<Tensor>(output_tensors.at(3)) : std::nullopt;
```

Comment:
```suggestion
    const auto& max_exp_avg_sq_out =
        this->amsgrad ? std::make_optional<Tensor>(output_tensors.at(3)) : std::nullopt;
```
`max_exp_avg_sq_out` is not null, if `amsgrad` is true.

Reply:
Applied